### PR TITLE
Add marrow utilities

### PR DESCRIPTION
**Pull request by dj-xps**
Still some things left to do.

The setup.cfg testing includes a bunch of --cov options, I have only tested locally with
```ini
[tool:pytest]
addopts = -r fEsxw test
```

`asyncore` should be replaced by `asyncio`, the use of `asyncore` is currently limited to a test with `DebuggingSMTPServer`.

The repository could probably use some extra cleaning, even if just some autoformatter.
Perhaps drop support for Python 2 altogether?

### CODE REVIEW

File: marrow/mailer/release.py [hunk 1]
```diff
@@ -4,11 +4,25 @@
...
+AuthorInfo = namedtuple(
+	'Author', (
+		'name',
+		'email',
+	),
+)
 
-author = namedtuple('Author', ['name', 'email'])("Alice Bevan-McGregor", 'alice@gothcandy.com')
+info = VersionInfo(4, 2, 0)
+version = f"{info.major}.{info.minor}.{info.micro}"
```
Comment (dj-xps) [inline marrow/mailer/release.py]: I am unsure how to do versioning for this repository properly, would appreciate all the opinions. :)

File: marrow/mailer/release.py [hunk 2]
```diff
@@ -4,11 +4,25 @@
...
+
+author = AuthorInfo("Alice Bevan-McGregor", 'alice@gothcandy.com')
 
 description = "A light-weight modular mail delivery framework for Python 2.7+, 3.3+, Pypy, and Pypy3."
```
Comment (dj-xps) [inline marrow/mailer/release.py]: Description is quite outdated too right now, not sure what it should be

File: setup.py [hunk 3]
```diff
@@ -21,13 +21,21 @@
...
 
-tests_require = ['pytest', 'pytest-cov', 'pytest-spec', 'pytest-flakes', 'coverage', 'transaction']
+tests_require = [
+	'pytest',
+	'pytest-cov',
+	'pytest-spec',
+	'pytest-flakes',
+	'coverage',
+	'transaction',
+	'py3dns',
```
Comment (dj-xps) [inline setup.py:24]: py3dns is added as test dependency, it used to be optional

File: setup.py [hunk 4]
```diff
@@ -58,13 +66,10 @@
...
 		include_package_data = True,
 		package_data = {'': ['README.textile', 'LICENSE.txt']},
-		namespace_packages = ['marrow'],
 		
 		# ## Dependency Declaration
 		
-		install_requires = [
-				'marrow.util < 2.0',
-			],
+		install_requires = ['python-magic'],
```
Comment (dj-xps) [inline setup.py:64]: To replace `imghdr`

File: test/test_validator.py [hunk 5]
```diff
@@ -92,79 +79,32 @@ def closure(domain, expect):
...
-		yield closure, domain, kind, expect
-
-
-def test_domain_validation():
-	try:
-		import DNS
-	except ImportError:
-		pytest.skip("PyDNS not installed.")
-
-	mock = DomainValidator(lookup_dns='mx')
```
Comment (dj-xps) [inline test/test_validator.py:129]: Deleted this test for now, as I did not have any live `mx` readily available

**Review by GertBurger — APPROVED** [review]

**Pull request merged**